core/ir/x86/instr.c

```diff
@@ -508,6 +508,15 @@ instr_branch_type(instr_t *cti_instr)
         return LINK_INDIRECT | LINK_CALL | LINK_FAR;
     case OP_call_far_ind: return LINK_INDIRECT | LINK_CALL | LINK_FAR;
     case OP_ret_far:
+    /* We don't mark sysenter and syscall as indirect branches because
+     * the user-mode DynamoRIO instrumentation does not need to treat them
+     * as such. sysexit and sysret are typically found in the kernel traces
+     * generated using other methods (like QEMU). It is useful to treat them
+     * as such to show proper continuity in the injected traces
```

Comment:
It may have been lost in the other content: but I was also suggesting that sysexit and sysret be generic indirect branches rather than returns, as they do not get their target from the stack nor give a hint to the hardware to predict as a return (presumably): so they don't really seem the same as other "returns".

Reply:
Sorry, missed that part. I removed LINK_RETURN. I also removed LINK_FAR because it seems to matter only for ibl lookups which isn't relevant for sysret/sysexit.

I'll make a similar change for the AArch64 eret in a later PR.

Reply:
eret: yes, agreed, it seems indirect branch would be better; it's kind of like sigreturn: in some sense it's a "return" but it's not in the sense that analysis tools expect with matching "call".